Mathlib/Data/ZMod/Units.lean

```diff
@@ -0,0 +1,23 @@
+/-
+Copyright (c) 2023 Moritz Firsching. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Firsching, Ashvni Narayanan, Michael Stoll
+-/
+import Mathlib.Data.ZMod.Basic
+
+
+/-!
+# Lemmas about units in `ZMod`.
+-/
+
+
+namespace ZMod
+
+variable {n m : ℕ}
+/-- `unitsMap` is a group homomorphism that maps units of `ZMod m` to units of `ZMod n` when `n`
+divides `m`. -/
+def unitsMap (hm : n ∣ m) : (ZMod m)ˣ →* (ZMod n)ˣ := Units.map (ZMod.castHom hm (ZMod n))
+
+lemma unitsMap_def (hm : n ∣ m) : ZMod.unitsMap hm = Units.map (ZMod.castHom hm (ZMod n)) := rfl
+
```

Comment:
Maybe add the `trans` lemma here:
```lean
lemma unitsMap_trans {m d : ℕ} (hm : n ∣ m) (hd : m ∣ d) :
  (unitsMap hm).comp (unitsMap hd) = unitsMap (dvd_trans hm hd)
```
and use that for the proof of `changeLevel_trans` later?

Reply:
Good idea! I did that (castHom_comp` was already there...).